### PR TITLE
refactor grib test for 6dp float change robustness

### PR DESCRIPTION
**Pull request by marqh**
remove an assertCML as numpy and the GRIB API  between them have subtly changed a the float resulting from a GRIB integer + scaling fed into a calculation of a points array based on start, step and count

### CODE REVIEW

**Comment by marqh** [comment]
ping @bjlittle 

this is my one outstanding unstable test for my new environment. this fix is a little crude, but it does maintain test coverage pretty well, imho and is not sensitive to the versions I am testing with

**Comment by marqh** [comment]
test failure
```
======================================================================
FAIL: test_polar_stereo_grib2 (iris.tests.integration.test_grib_load.TestBasicLoad)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/net/home/h04/itmh/scitools/iris/lib/iris/tests/integration/test_grib_load.py", line 107, in test_polar_stereo_grib2
    self.assertCML(cube, ("grib_load", "polar_stereo_grib2.cml"))
  File "/net/home/h04/itmh/scitools/iris/lib/iris/tests/__init__.py", line 384, in assertCML
    self._check_same(xml, reference_path)
  File "/net/home/h04/itmh/scitools/iris/lib/iris/tests/__init__.py", line 471, in _check_same
    type_comparison_name)
  File "/net/home/h04/itmh/scitools/iris/lib/iris/tests/__init__.py", line 221, in _assert_str_same
    self.fail("%s do not match: %s\n%s" % (type_comparison_name, reference_filename, diff))
AssertionError: CML do not match: /net/home/h04/itmh/scitools/iris/lib/iris/tests/results/grib_load/polar_stereo_grib2.cml
--- Reference
+++ Test result
@@ -23 +23 @@
-		-216.145905259]" shape="(200,)" standard_name="projection_y_coordinate" units="Unit('m')" value_type="float64">
+		-216.145905261]" shape="(200,)" standard_name="projection_y_coordinate" units="Unit('m')" value_type="float64">


```

this is a coordinate points array, calculated from a GRIB encoding of start and step in integer plus offset

we get a numerical difference of 2e-9, a factor of  1.000000000009

I can't see how this is mission critical

File: lib/iris/tests/integration/test_grib_load.py [hunk 1]
```diff
@@ -104,7 +104,15 @@ def test_polar_stereo_grib2(self):
         cube = iris.load_cube(tests.get_data_path(
             ("GRIB", "polar_stereo",
              "CMC_glb_TMP_ISBL_1015_ps30km_2013052000_P006.grib2")))
-        self.assertCML(cube, ("grib_load", "polar_stereo_grib2.cml"))
```
Comment (bjlittle) [inline lib/iris/tests/integration/test_grib_load.py]: @marqh Do we need to remove the cml file `iris/tests/results/grib_load/polar_stereo_grib2.cml` ?

What about checking the integrity of the other cube metadata? i.e. there's a `forecast_period`, `forecast_reference_time`, `pressure` and `time` coordinates?

We could could be more rigorous in checking the CRS metadata is equivalent to `<stereographic central_lat="90.0" central_lon="249.0" ellipsoid="GeogCS(6371229.0)" false_easting="0.0" false_northing="0.0" true_scale_lat="60.0"/>`

**Comment by marqh** [comment]
review actions:

removed unused CML
add crs checks
change test name to be explicily about grid definition

ping @bjlittle

**Pull request merged**